.cascade-code/Chapman.edu/_components/omni_nav_v2/omni_nav.vtl

```diff
@@ -0,0 +1,812 @@
+  [ 'support-chapman', 'Support Chapman', 'default', '/' ],
+  [ 'communication', 'SCHOOL OF COMM', 'soc', '/communication/' ]
+])
+
+##
+## Vars
+##
+## Cascade provides a $currentPage object to every format.
+#set ( $pagePath = $currentPage.path )
+
+## This will be set by data macros below.
+#set ( $pageUmbrellaCategory = [] )
+#set ( $isUmbrellaCategory = false )
+#set ( $umbrellaDirectory = '' )
+#set ( $umbrellaDirectoryPath = '' )
+#set ( $umbrellaLabel = '' )
+#set ( $umbrellaClass = '' )
+#set ( $umbrellaHref = '' )
+#set ( $umbrellaBrandLabel = '' )
+#set ( $umbrellaAssets = [] )
+#set ( $brandClass = '' )
+
+
+##
+## Var Macros
+##
+#macro ( setVars )
+  #setPageUmbrellaCategory()
+
+  #set ( $isUmbrellaCategory = $pageUmbrellaCategory.size() > 0 )
+
+  #if ( $isUmbrellaCategory )
+    #set ( $umbrellaDirectory = $pageUmbrellaCategory[0] )
+
+    ## These 2 were originally different values and could differ in the future.
+    #set ( $umbrellaLabel = $pageUmbrellaCategory[1] )
+    #set ( $umbrellaBrandLabel = $pageUmbrellaCategory[1] )
+    
+    #set ( $umbrellaClass = $pageUmbrellaCategory[2] )
+    #set ( $umbrellaHref = $pageUmbrellaCategory[3] )
+    #set ( $umbrellaDirectoryPath = "/${umbrellaDirectory}" )
+    #setUmbrellaAssets()
+    #set ( $brandClass = 'branded' )
+  #else
+    #set ( $brandClass = 'unbranded' )
+  #end
+#end
+
+#macro ( setPageUmbrellaCategory )
+  ## Is current path under one of specified $umbrellaCategories?
+  #foreach ( $umbrellaCategory in $umbrellaCategories )
+    #set ( $dirName = $umbrellaCategory[0] )
+
+    ## Define substr range to compare umbrella dir with current page path.
+    ## pagePath does not include leading /.
+    #set ( $startSubStr = 0 )
+    #set ( $endSubStr = $dirName.length() )
+
+    ## Make sure endSubStr not long than pagePath else will throw exception
+    #if ( $endSubStr > $pagePath.length() )
+      #set ( $endSubStr = $pagePath.length() )
+    #end
+
+    #set ( $pagePathDir = $pagePath.substring($startSubStr, $endSubStr) )
+
+    ## Is current page directory $pagePathDir under this $navDirectory? If so,
+    ## we found our $navDirectory.
+    #if ( $pagePath.contains($dirName) && $pagePathDir == $dirName )
+      #set ( $pageUmbrellaCategory = $umbrellaCategory )
+    #end
+  #end
+#end
+
+#macro ( setUmbrellaAssets )
+  ## umbrellaAssets equal Umbrella directory's visible (non-index) children. These
+  ## are used in OffCanvasNav secondary menu. This must be called after
+  ## $umbrellaDirectoryPath is set.
+  #set ( $umbrellaFolder = $_.locateFolder($umbrellaDirectoryPath) )
+
+  #foreach( $childAsset in $umbrellaFolder.children )
+    #set ( $metaHide = $childAsset.metadata.getDynamicField('Hide from navigation').value )
+
+    ## Skip if hidden or is index page.
+    #if ( $metaHide != 'Yes' && $childAsset.name != 'index' )
+      #set ( $discard = $umbrellaAssets.add($childAsset) )
+    #end
+  #end
+#end
+
+
+##
+## Template Macros
+##
+#macro ( buildOmniNav )
```

Comment:
Main macro that builds the OmniNav. Macros for each part of OmniNav are organized into sections below.